src/Catalyst.Core.Modules.Kvm/DeltaExecutor.cs

```diff
@@ -0,0 +1,519 @@
+        public void Execute(Delta stateUpdate, ITxTracer txTracer) { Execute(stateUpdate, txTracer, false); }
+
+        /// <summary>
+        ///     This method calculates the basic gas cost of the <paramref name="entry" />
+        ///     (excluding the cost of the VM execution).
+        ///     The intrinsic cost is calculated as 21000 base cost + the entry data cost that is a function
+        ///     of the number of bytes of data (and the cost of a single byte may differ between 0 bytes and non-0 bytes.
+        /// </summary>
+        /// <param name="entry"></param>
+        /// <param name="releaseSpec"></param>
+        /// <returns>Total intrinsic cost of the <paramref name="entry" /></returns>
+        public ulong CalculateIntrinsicGas(ContractEntry entry, IReleaseSpec releaseSpec)
+        {
+            ulong result = GasCostOf.Transaction; // the basic entry cost
+            if (entry.Data != null)
+            {
+                // here is the difference between the 0 bytes and non-zero bytes cost
+                // justified by a better compression level of zero bytes
+                long txDataNonZeroGasCost = releaseSpec.IsEip2028Enabled ? GasCostOf.TxDataNonZeroEip2028 : GasCostOf.TxDataNonZero;
+                int length = entry.Data.Length;
+                for (int i = 0; i < length; i++) result += entry.Data[i] == 0 ? GasCostOf.TxDataZero : (ulong) txDataNonZeroGasCost;
+            }
+
+            if (entry.IsValidDeploymentEntry && releaseSpec.IsEip2Enabled) result += GasCostOf.TxCreate;
+
+            return result;
+        }
+
+        [Todo(Improve.MissingFunctionality, "We need to agree on delta to state mapping details")]
+        private StateUpdate ToStateUpdate(Delta delta)
+        {
+            StateUpdate result = new StateUpdate
+            {
+                Difficulty = 1,
+                Number = 1,
+                Timestamp = (UInt256) delta.TimeStamp.Seconds,
+                GasLimit = (long) delta.GasLimit,
+                /* here we can read coinbase entries from the delta
+                   but we need to decide how to split fees and which one to pick for the KVM */
+                GasBeneficiary = Address.Zero,
+                GasUsed = 0L
+            };
+
+            return result;
+        }
+
+        private Address GetAccountAddress(ByteString publicKeyByteString)
+        {
+            if (publicKeyByteString == null || publicKeyByteString.IsEmpty) return null;
+
+            IPublicKey publicKey = _cryptoContext.GetPublicKeyFromBytes(publicKeyByteString.ToByteArray());
+            return publicKey.ToKvmAddress();
+        }
+
+        private void QuickFail(ContractEntry entry, ExecutionEnvironment env, ITxTracer txTracer)
+        {
+            // here we need to propagate back to Delta
+            // env.CurrentBlock.GasUsed += gasLimit;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4350015)